compiler/rustc_ast/src/ast.rs

```diff
@@ -2028,10 +2028,12 @@ pub enum InlineAsmOperand {
 #[derive(Clone, Encodable, Decodable, Debug)]
 pub struct InlineAsm {
     pub template: Vec<InlineAsmTemplatePiece>,
+    pub template_strs: Box<[(Symbol, Option<String>, Span)]>,
```

Comment:
This can just be a `Vec`, it's not like any of the other `Vec`s in here are ever modified either.

Reply:
With the above `is_local` bool, a `Vec` was increasing the total size of the `ItemKind` enum and causing an assert to fail.  While I could have bumped the size of that assert, it seemed like a bad idea to increase the size of that enum for a very infrequent use case for performance concerns.  The `Box<[]>` was meant to reduce that size since I didn't need to be able to mutate it once it was created.

However with the removal of `is_local`, I can probably make that a `Vec` if you think that's clearer/better.